### ISSUE TITLE
Switch back to scan-free softmax

### ISSUE THREAD

**Issue by elanmart** [fchollet/keras#1185]
@fchollet Is there any reason why we can't have a softmax that uses reshape instead of scan?

For my applications it's about ~30ms (251 vs 287) faster for a 128 batchsize with 10k vocabulary. It's not much compared to the rest of the net, but it's a free performance gain.

For applications like #1168, where the number of steps is big compared to number of classes, it's twice slower with scan (16ms vs 7ms)

**Comment by fchollet** [comment]
Sure, we could. The change was driven by code simplicity, I didn't expect much of a performance change.

Would you like to submit a PR?

**Comment by elanmart** [comment]
Sure, will do it shortly then.

**Comment by EderSantana** [comment]
@elanmart just so you know, I had recent problems with reshaping. Under TF it forces us to define batch size.
